### ISSUE TITLE
6to5 transpiler with optional 'selfContained' transformer

### ISSUE THREAD

**Issue by u9520107** [jspm/jspm-cli#508]
Not sure if jspm's problem, but I am having problems with loading es6 modules if I set the transpiler to use the 'selfContained' transformer.
```
<html>
  <head>
    <script src="/jspm_packages/system.js"></script>
    <script src="/jspm_packages/config.js"></script>
    <script> System.to5Options = { optional: ['selfContained']};
       System.import('test.js');
   </script>
  ...</html>
```
Where test.js is simply:
```
export default function * () {}
```

The transpiled code has lines as such:
```
  ...
 execute: function () {
      _export("default", _regeneratorRuntime.mark(function callee$0$0() {
```

However, when I try to trace the code, _regenertorRuntime does not have a function called mark.
Instead, it has a default property, where mark is a member function of that default property.
I know 6to5 sugers mod.default to mod with some interop helpers, but that doesn't seem to happen properly in this case...

**Comment by u9520107** [comment]
I stopped using the self-contained option, and loads the 6to5-runtime and 6to5-polyfill instead and things seems to work fine. Not exactly use why though. I wish I had more time on this matter... 
Feel free to close if it's not a big issue, I'll stick with the runtime and polyfill for now.

**Comment by guybedford** [comment]
When you use the self-contained option does it contain a dependency on `regenerator`?

**Comment by u9520107** [comment]
Think so... I used a lot of generator function throughout my projects.
But then _regenerator.default does contained the wanted object, except the transpiled code assumed that it should've been _regenerator instead of calling the default object.

**Comment by guybedford** [comment]
So you mean it's accessing the default export instead of `_regenerator` directly? Could you copy in the compiled output example of this?

**Comment by u9520107** [comment]
Err.. The transpiled code tried _regenerator.mark directly. But at runtime _regenerator appears to only have a default prop object, which in turn contains the mark function.

-----Original Message-----

**Comment by guybedford** [comment]
@u9520107 ok this is a 6to5 bug then we should get them to use the `.default` access.

**Comment by guybedford** [comment]
Created https://github.com/6to5/6to5/issues/774.

**Comment by guybedford** [comment]
Already fixed in https://github.com/6to5/6to5/commit/8a5da5716278a74a53691c77475a88773cd4ae4c!

**Comment by u9520107** [comment]
Awesome!
I'll see if I can test this out in a few days time...

**Comment by sebmck** [comment]
Released as of 3.6.3

**Comment by guybedford** [comment]
Should be resolved now in jspm@0.14.
